zulipterminal/ui_tools/boxes.py

```diff
@@ -650,21 +704,12 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     else:
                         header.focus_col = self.FOCUS_HEADER_BOX_STREAM
                 else:
-                    self.update_recipient_emails(self.to_write_box)
-                    invalid_emails = [
-                        recipient_email
-                        for recipient_email in self.recipient_emails
-                        if not self.model.is_valid_private_recipient(
-                            recipient_email
-                        )
-                    ]
-
-                    if invalid_emails:
-                        invalid_emails_error = (
-                            f"Invalid recipient(s) - {', '.join(invalid_emails)}"
-                        )
-                        self.view.controller.report_error(invalid_emails_error)
+                    all_valid = self.tidy_valid_recipients_and_notify_invalid_ones(
+                        self.to_write_box
+                    )
+                    if not all_valid:
                         return key
+                    self.update_recipient_emails(self.to_write_box)
```

Comment:
Is there some change in behavior by having this after instead of earlier? Does it warrant a mention?

Reply:
It ensures that the emails extracted are valid for sure, I've added a comment for the same.